types/net_matches.d.ts

```diff
@@ -0,0 +1,12 @@
+import { NetAllFields } from './net_fields';
+import { TriggerTypes } from './net_trigger';
+
+export type Matches = { [s in T]?: string };
+export type ToMatches<Fields> = Record<keyof Fields, string>;
+
+export type NetAllMatches = {
```

Comment:
This is just a naming suggestion, but feel free to disagree.  I feel like `NetAllMatches` is now used in a bunch of places, but is kind of a long name when paired with square brackets.  What do you think about `NetMatches` and dropping the `All` out of it?

Feel free to disagree and we can go with what's here.  Just a drive-by thought.

Reply:
There's a bit of a naming schema going on here, these are all named with the same schema:

```ts
NetAllMatches
NetAllFields
NetAllParams
NetAllProps
```

I feel like if we rename one of them, we should rename all of them?

Reply:
Went ahead and renamed them all, feel free to resolve if that's good for you.